### PR TITLE
pydeck/jupyter-widget: Call mergeConfiguration again after addCustomLibraries complete

### PR DESCRIPTION
**Pull request by ibgreen**
For #4103 

#### Background
- As discussed in #4375 there is an issue when the jupyter widget initializes using a synchronous API but the custom library script loading happens asynchronously. 

#### Change List
- Wait for script loading to complete

### CODE REVIEW

File: modules/jupyter-widget/src/create-deck.js [hunk 1]
```diff
@@ -45,32 +45,57 @@ const jsonConverter = new deck.JSONConverter({
...
+
+  function onEachFinish() {
+    if (Object.keys(loaded).length === customLibraries.length) {
+      // when all libraries loaded
+      if (typeof onComplete === 'function') onComplete();
+    }
   }
+
+  customLibraries.forEach(({libraryName, resourceUri}) => {
+    if (window[libraryName]) {
```
Comment (Pessimistress) [inline modules/jupyter-widget/src/create-deck.js]:
```suggestion
    if ('libraryName' in window) {
```

File: modules/jupyter-widget/src/create-deck.js [hunk 2]
```diff
@@ -45,32 +45,57 @@ const jsonConverter = new deck.JSONConverter({
...
+function addCustomLibraries(customLibraries, onComplete) {
   if (!customLibraries) {
     return;
   }
-  for (const {libraryName, resourceUri} of customLibraries) {
-    loadExternalLibrary({libraryName, resourceUri});
+  const loaded = {};
+
+  function onEachFinish() {
+    if (Object.keys(loaded).length === customLibraries.length) {
```
Comment (Pessimistress) [inline modules/jupyter-widget/src/create-deck.js]: What if `addCustomLibraries` is called multiple times with different parameters? It's more robust to set `loaded[libraryName] = false` before loading and check `Object.keys(loaded).every(libraryName => loaded[libraryName])` after each load.

File: modules/jupyter-widget/src/create-deck.js [hunk 3]
```diff
@@ -96,6 +123,16 @@ function createDeck({
       container
     });
 
+    const onComplete = () => {
+      const newProps = jsonConverter.convert({layers: layerToLoad});
+      const mergeLayers = [...props.layers, ...newProps.layers].filter(l => l);
```
Comment (Pessimistress) [inline modules/jupyter-widget/src/create-deck.js]: - Add some comments on why we need to do this
- Doesn't this change the order of layers?

**Comment by ibgreen** [comment]
@ajduberstein Would be good to get this into a `0.3.0` build soon for further end-to-end testing.

**Pull request merged**